test/coverageLib/FileCoverage.test.js

```diff
@@ -0,0 +1,414 @@
+    };
+}
+
+function identifier(value, locationNum = 1) {
+    return token(Lexeme.Identifier, value, locationNum);
+}
+
+function locationEqual(loc1, loc2) {
+    return (
+        loc1.start.line === loc2.start.line &&
+        loc1.start.column === loc2.start.column &&
+        loc1.end.line === loc2.end.line &&
+        loc1.end.column === loc2.end.column
+    );
+}
+
+describe("FileCoverage", () => {
+    beforeEach(() => {
+        fg.sync.mockImplementation(() => {
+            return [
+                "baseComponent.xml",
+                "extendedComponent.xml",
+                "scripts/baseComp.brs",
+                "scripts/extendedComp.brs",
+                "scripts/utility.brs",
+            ];
+        });
+        fs.readFile.mockImplementation((filename, _, cb) => {
+            resourcePath = path.join(__dirname, "resources", filename);
+            realFs.readFile(resourcePath, "utf8", (err, contents) => {
+                cb(/* no error */ null, contents);
+            });
+        });
+    });
+
+    afterEach(() => {
+        fg.sync.mockRestore();
+        fs.readFile.mockRestore();
+    });
+
+    // describe("integration", () => {
+    //     it("runs", () => {
+    //         expect(true).toEqual(true);
+    //     });
+    // });
+
+    // describe("getCoverage", () => {
+    //     it("runs", () => {
+    //         expect(true).toEqual(true);
+    //     });
+    // });
+
+    describe("statements", () => {
+        function checkSimpleStatement(statement, numHits = 1, expectedNumStatements = 1) {
+            let fileCoverage = new FileCoverage("path/to/file");
+            fileCoverage.execute(statement);
+
+            for (let i = 0; i < numHits; i++) {
+                fileCoverage.logHit(statement);
+            }
+
+            let coverageResults = fileCoverage.getCoverage();
+            expect(Object.keys(coverageResults.branches).length).toEqual(0);
+            expect(Object.keys(coverageResults.functions).length).toEqual(0);
+
+            let statementKeys = Object.keys(coverageResults.statements);
+            expect(statementKeys.length).toEqual(expectedNumStatements);
+
+            let statementCoverage = coverageResults.statements[statementKeys[0]];
+            expect(locationEqual(statementCoverage.location, statement.location)).toBeTruthy();
+            expect(statementCoverage.hits).toEqual(numHits);
+        }
+
+        describe("Correct number of hits", () => {
+            it("0 hits", () => {
+                checkSimpleStatement(new Stmt.Expression(new Expr.Literal(new Int32(1))), 0);
+            });
+
+            it("1 hit", () => {
+                checkSimpleStatement(new Stmt.Expression(new Expr.Literal(new Int32(1))), 10);
+            });
+
+            it("10 hits", () => {
+                checkSimpleStatement(new Stmt.Expression(new Expr.Literal(new Int32(1))), 10);
+            });
+        });
+
+        it("Assignment", () => {
```

Comment:
super nit: consider using `test()` for these since the names of the tests don't form sentences/statements/behaviors/etc.